src/project/manifest/mod.rs

```diff
@@ -308,13 +308,25 @@ impl Manifest {
         dep: &PackageName,
         spec_type: SpecType,
         platform: Option<Platform>,
+        feature_name: Option<&FeatureName>,
     ) -> miette::Result<(PackageName, NamelessMatchSpec)> {
-        get_toml_target_table(&mut self.document, platform, spec_type.name())?
+        get_toml_target_table(&mut self.document, platform, feature_name, spec_type.name())?
             .remove(dep.as_normalized())
             .ok_or_else(|| {
-                let table_name = match platform {
-                    Some(platform) => format!("target.{}.{}", platform.as_str(), spec_type.name()),
-                    None => spec_type.name().to_string(),
+                let table_name = match (platform, feature_name) {
```

Comment:
I think it makes sense to move this to a function

Reply:
Changed in https://github.com/prefix-dev/pixi/pull/680/commits/5556e0b53355a9bfd6e7fd227ac22cea19063cfc